store/postgres/src/sharded_store.rs

```diff
@@ -649,6 +652,57 @@ impl StoreTrait for ShardedStore {
         let info = store.subgraph_info(id)?;
         Ok(info.network)
     }
+
+    fn version_info(&self, version: &str) -> Result<VersionInfo, StoreError> {
+        let primary = self.primary_conn()?;
```

Comment:
Should all of the store operations in this fn be done atomically within a single transaction?

Reply:
There's not much we could gain there: we need to first consult the primary for details about the subgraph, then look up information about the deployment in the deployment's shard. The only other thing we get from that shard is `store.subgraph_info`, which is immutable info that is read once and cached for the lifetime of the store instance. To finish, we look up the chain head pointer - right now, that lives in the primary, but will soon move to (possibly) it's own shard.

So unfortunately, there's not much we can gain in terms of atomicity